### PR TITLE
add_settings_for_operation

### PR DESCRIPTION
**Pull request by Tkachenko01001**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -25,7 +26,7 @@ function App() {
   )
   const dispatch = useDispatch<AppDispatch>()
 
-  const token = useSelector((state: RootState) => state.auth.accessToken)
+  const accessToken = Cookies.get('token')
```
Comment (pqoqubbw) [inline src/App.tsx]: `token` should be a variable like `const COOKIE_TOKEN_NAME = "token"`

File: src/App.tsx [hunk 2]
```diff
@@ -36,12 +37,12 @@ function App() {
   }, [allCategories, navigate])
 
   useEffect(() => {
-    if (token && !isLoggedIn) {
+    if (accessToken && !isLoggedIn) {
       dispatch(
-        currentUser({ accessToken: token, operationType: 'currentUser' })
+        currentUser({ accessToken: accessToken, operationType: 'currentUser' })
```
Comment (pqoqubbw) [inline src/App.tsx]:
```suggestion
        currentUser({ accessToken, operationType: 'currentUser' })
```

File: src/axios/settings.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios'
+import Cookies from 'js-cookie'
+
+const axiosInstance = axios.create({
+  baseURL: 'https://www.test-store.shop/'
```
Comment (pqoqubbw) [inline src/axios/settings.ts]: should be as a `const`

File: src/axios/settings.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+})
+
+axiosInstance.interceptors.request.use(
+  (config) => {
+    const accessToken = Cookies.get('token')
```
Comment (pqoqubbw) [inline src/axios/settings.ts]: use variable

File: src/axios/settings.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (accessToken) {
+      if (config.headers) config.headers.Authorization = `Bearer ${accessToken}`
+    }
```
Comment (pqoqubbw) [inline src/axios/settings.ts]:
```suggestion
    if (accessToken && config.headers) {
      config.headers.Authorization = `Bearer ${accessToken}`
    }
```

File: src/components/Layout/header/header-nav/DropdownList.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+type Props = {
+  allCategories: CategoriesModel[]
+}
+
+const DropdownList: FC<Props> = ({ allCategories }) => {
+  return (
+    <ul className={styles.dropdownList}>
+      {allCategories &&
+        allCategories.map((category) => {
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/DropdownList.tsx]:
```suggestion
      {allCategories?.map((category) => {
```

File: src/components/Layout/header/header-nav/HeaderNavList.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import { CategoriesModel } from '../../../../types/models/CategtoriesModel'
+import CrmLine from './CrmLink'
+
+type Props = {
+  allCategories: CategoriesModel[]
+}
+
+const HeaderNavList: FC<Props> = ({ allCategories }) => {
+  const location = useLocation()
+  const homePath = location.pathname === '/'
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/HeaderNavList.tsx]: Use the right name convention for boolean, ex `isHomePath`

File: src/components/Layout/header/header-nav/HeaderNavList.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <Link
+          to={`catalog/${allCategories && allCategories[0].id}/0`}
+          className={styles.listNavLink}
+        >
+          Каталог
+        </Link>
+        <DropdownList allCategories={allCategories} />
+      </li>
+      <li className={styles.listNavLine}>
+        <Link to={'/review'} className={styles.listNavLink}>
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/HeaderNavList.tsx]:
```suggestion
        <Link to="/review" className={styles.listNavLink}>
```

File: src/components/Layout/header/header-nav/headerNav.tsx [hunk 9]
```diff
@@ -1,25 +1,21 @@
...
   )
-  const dispatch = useDispatch<AppDispatch>()
 
-  useEffect(() => {
-    if (!allCategories) {
-      dispatch(fetchAllCategories({ operationType: 'fetchAllCategories' }))
-    }
-  }, [allCategories])
+  const location = useLocation()
+  const homePath = location.pathname === '/'
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/headerNav.tsx]: same

File: src/components/Layout/header/header-nav/headerNav.tsx [hunk 10]
```diff
@@ -30,61 +26,15 @@ const HeaderNav = () => {
     }
   }, [homePath, location])
 
+  useEffect(() => {
+    if (!allCategories) {
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/headerNav.tsx]: maybe you meant `!allCategories?.length`?

File: src/components/privste-route/index.tsx [hunk 11]
```diff
@@ -1,15 +1,14 @@
...
-import { useAuth } from '../../hooks/use-auth'
-
 type Props = {
   component: ReactNode
   redirectTo: string
 }
 
 const PrivateRoute: FC<Props> = ({ component, redirectTo }) => {
-  const { accessToken } = useAuth()
+  const accessToken = Cookies.get('token')
```
Comment (pqoqubbw) [inline src/components/privste-route/index.tsx]: use token variable

File: src/redux/authentication/operation.ts [hunk 12]
```diff
@@ -63,6 +65,9 @@ export const login = createAsyncThunk<LoginResponse, LoginParams>(
           }
         }
       )
+
+      Cookies.set('token', response.data.access_token, { expires: 7 })
```
Comment (pqoqubbw) [inline src/redux/authentication/operation.ts]: same

File: src/redux/authentication/operation.ts [hunk 13]
```diff
@@ -106,13 +107,10 @@ type LogOutParams = {
...
-  async ({ accessToken }, thunkAPI) => {
+  async (_, thunkAPI) => {
     try {
-      const response = await axios.post('api/auth/logout', null, {
-        headers: {
-          Authorization: `Bearer ${accessToken}`
-        }
-      })
+      const response = await axiosInstance.post('api/auth/logout', null)
+      Cookies.remove('token')
```
Comment (pqoqubbw) [inline src/redux/authentication/operation.ts]: same

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Tkachenko01001) [reply src/App.tsx]: done

File: src/App.tsx (hunk 2 above)
Comment (Tkachenko01001) [reply src/App.tsx]: done

File: src/axios/settings.ts [hunk 14]
```diff
@@ -1,16 +1,19 @@
 import axios from 'axios'
 import Cookies from 'js-cookie'
 
+const BASE_URL = 'https://www.test-store.shop/'
+const COOKIE_TOKEN_NAME = 'token'
```
Comment (pqoqubbw) [inline src/axios/settings.ts]: just export already created variable

File: src/components/privste-route/index.tsx [hunk 15]
```diff
@@ -2,13 +2,15 @@ import { Navigate } from 'react-router-dom'
 import Cookies from 'js-cookie'
 import { FC, Fragment, ReactNode } from 'react'
 
+const COOKIE_TOKEN_NAME = 'token'
```
Comment (pqoqubbw) [inline src/components/privste-route/index.tsx]: same

File: src/redux/authentication/operation.ts [hunk 16]
```diff
@@ -5,6 +5,8 @@ import { TokenModel, UserResponse } from '../../types'
 import { OperationType, SignUpFormData } from './slice'
 import Cookies from 'js-cookie'
 
+const COOKIE_TOKEN_NAME = 'token'
```
Comment (pqoqubbw) [inline src/redux/authentication/operation.ts]: same

**Review by pqoqubbw — CHANGES_REQUESTED** [review]
maybe it will be better to create custom hook like `useCookie` which allow you to get and set your value

File: src/App.tsx [hunk 17]
```diff
@@ -28,7 +26,10 @@ function App() {
   )
   const dispatch = useDispatch<AppDispatch>()
 
-  const accessToken = Cookies.get(COOKIE_TOKEN_NAME)
+  const { getToken } = useCookieMenager()
```
Comment (pqoqubbw) [inline src/App.tsx]: typo, manager

File: src/components/Layout/header/header-nav/CrmLink.tsx [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+import { Link } from 'react-router-dom'
+import { useAuth } from '../../../../hooks/use-auth'
+import styles from '../Header.module.scss'
+
+const CrmLink = () => {
+  const { user } = useAuth()
+
+  return (
+    <React.Fragment>
+      {user && (user.role === 'admin' || user.role === 'moderator') && (
```
Comment (pqoqubbw) [inline src/components/Layout/header/header-nav/CrmLink.tsx:11]:
```suggestion
      {(user?.role === 'admin' || user?.role === 'moderator') && (
```

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

**Review by pqoqubbw — APPROVED** [review]

**Pull request merged**